packages/grafana-data/src/field/fieldOverrides.ts

```diff
@@ -209,19 +208,19 @@ function setDynamicConfigValue(config: FieldConfig, value: DynamicConfigValue, c
   const remove = val === undefined || val === null;
 
   if (remove) {
-    if (value.isCustom && config.custom) {
-      delete config.custom[item.path];
+    if (item.isCustom && config.custom) {
+      delete config.custom[item.path.split('.')[0]];
```

Comment:
not sure I understand this, introducing path was to avoid splitting id

Reply:
and it looks strange, is custom.nestedProp.value is undefined, wht is custom.nestedProp deleted?

Reply:
but you can't `delete` based on path (if nested).

Reply:
then I think we should not support nested props, or I don't know, this looks like it does not work either.

So this remove clause, do we need this?  what are we removing here? where is the thing we are removing coming from? are we removing defaults?

Reply:
not sure we need the remove/delete handling here: 

only when we modify the persisted model: 
https://github.com/grafana/grafana/blob/master/public/app/features/dashboard/components/PanelEditor/FieldConfigEditor.tsx#L125 

Wonder if we should share some code between this function and the function used by the editor to remove & set default values

Reply:
yeah, we probably could, i'll followup on this ltr